internal/attestation/renderer/renderer.go

```diff
@@ -87,15 +106,85 @@ func (ab *AttestationRenderer) Render() (*dsse.Envelope, error) {
 		return nil, err
 	}
 
-	signedAtt, err := ab.signer.SignMessage(bytes.NewReader(rawStatement))
+	signedAtt, err := ab.dsseSigner.SignMessage(bytes.NewReader(rawStatement))
 	if err != nil {
 		return nil, fmt.Errorf("signing message: %w", err)
 	}
 
-	var dseeEnvelope dsse.Envelope
-	if err := json.Unmarshal(signedAtt, &dseeEnvelope); err != nil {
+	var dsseEnvelope dsse.Envelope
+	if err := json.Unmarshal(signedAtt, &dsseEnvelope); err != nil {
 		return nil, err
 	}
 
-	return &dseeEnvelope, nil
+	if ab.bundlePath != "" {
+		// Create sigstore bundle for the contents of this attestation
+		bundle, err := ab.envelopeToBundle(dsseEnvelope)
+		if err != nil {
+			return nil, fmt.Errorf("loading bundle: %w", err)
+		}
+		json, err := protojson.Marshal(bundle)
+		if err != nil {
+			return nil, fmt.Errorf("marshalling bundle: %w", err)
+		}
+		ab.logger.Info().Msg(fmt.Sprintf("generating Sigstore bundle %s", ab.bundlePath))
+		err = os.WriteFile(ab.bundlePath, json, 0600)
+		if err != nil {
+			return nil, fmt.Errorf("writing bundle: %w", err)
+		}
+	}
+
+	return &dsseEnvelope, nil
+}
+
+func (ab *AttestationRenderer) envelopeToBundle(dsseEnvelope dsse.Envelope) (*protobundle.Bundle, error) {
+	// DSSE Envelope is already base64 encoded, we need to decode to prevent it from being encoded twice
+	payload, err := base64.StdEncoding.DecodeString(dsseEnvelope.Payload)
+	if err != nil {
+		return nil, fmt.Errorf("decoding: %w", err)
+	}
+	bundle := &protobundle.Bundle{
+		MediaType: "application/vnd.dev.sigstore.bundle+json;version=0.3",
+		Content: &protobundle.Bundle_DsseEnvelope{DsseEnvelope: &dsse2.Envelope{
+			Payload:     payload,
+			PayloadType: dsseEnvelope.PayloadType,
+			Signatures: []*dsse2.Signature{
+				{
+					Sig:   []byte(dsseEnvelope.Signatures[0].Sig),
+					Keyid: dsseEnvelope.Signatures[0].KeyID,
+				},
+			},
+		}},
+		VerificationMaterial: &protobundle.VerificationMaterial{},
+	}
+
+	// check type of wrapped signer
+	switch v := ab.signer.(type) {
+	case *cosign.Signer:
+		pk, err := signature.PublicKeyPem(v)
+		if err != nil {
+			return nil, fmt.Errorf("getting public key: %w", err)
+		}
+
+		bundle.VerificationMaterial.Content = &protobundle.VerificationMaterial_PublicKey{PublicKey: &v12.PublicKeyIdentifier{
+			Hint: string(pk),
+		}}
+	case *chainloopsigner.Signer:
```

Comment:
could this be encapsulated in the signer interface itself?

basically something like `(s ChainloopSigner) VerificationMaterial`?

Reply:
Yes I was thinking that. Let me give it a look

Reply:
Please mark the Pr as draft if you plan to do any changes. Thanks!

Reply:
I'll do it as part of a separate PR, as it requires defining a new interface and moving some packages.